sdk/azidentity/azidentity.go

```diff
@@ -151,6 +151,14 @@ func doForClient(client *azcore.Client, r *http.Request) (*http.Response, error)
 			return nil, err
 		}
 	}
+
+	h := req.Raw().Header
+	for key, val := range r.Header {
+		for i := range val {
+			h.Add(key, val[i])
+		}
+	}
```

Comment:
I will accept this suggestion, but let me just ask - are you concerned that we could have conflicting values for some headers or are you concerned that we might loose the new req header value for the old one? we will still have the new request headers persisted since I am doing `.Add` not `.Set`, and, if I'm not mistaken, usually if there are multiple values in the header, the first one is used (that's what `http.Header.Get()` does by default)

so the only concern I might have is that new request has a header `{"doSomething": "true"}` and then the original request had header `{"doSomething": "false"}` and we might end up with `{"doSomething": "true", "false"}`

Reply:
> are you concerned that we could have conflicting values for some headers or are you concerned that we might loose the new req header value for the old one? 

Both. I don't want to modify any header set by the Azure SDK.

> we will still have the new request headers persisted since I am doing `.Add` not `.Set`

I'm concerned that sending multiple values could change server behavior because I don't know whether the relevant RFCs stipulate how to interpret them and I don't expect consistent behavior across servers. Unconditionally adding will also duplicate values for common headers e.g. `Content-Type: ['application/json', 'application/json']`

Actually `Set` would be simpler here:

```suggestion
	// copy headers to the new request, ignoring any for which the new request has a value
	for key, val := range r.Header {
		if _, has := h[key]; !has {
			h.Set(key, val)
		}
	}
```

Reply:
accepted

Reply:
Whoops, my bad, I assumed `Set` would preserve all values but apparently it doesn't. In that case I suppose we must `Add` to avoid discarding values when a key has more than one (probably this never comes up but it would be a subtle bug)

Reply:
so, should I go back to

```go
	h := req.Raw().Header
	for key, vals := range r.Header {
		for _, val := range vals {
			h.Add(key, val)
		}
	}
```

or do you want to avoid modifying headers set by azure-sdk, but do not destroy the ones set by msal

```go
	h := req.Raw().Header
	for key, vals := range r.Header {
		if _, has := h[key]; !has {
			for _, val := range vals {
				h.Add(key, val)
			}
		}
	}
```

Reply:
Option 2 please. For all headers set on `r`, if the header isn't set on `req`, copy all values from `r`.

Reply:
added